src/gt4py/next/iterator/transforms/global_tmps.py

```diff
@@ -103,20 +169,58 @@ def _transform_by_pattern(
             ) = type_info.apply_to_primitive_constituents(
                 type_info.extract_dtype,
                 tmp_expr.type,
-                tuple_constructor=lambda *elements: tuple(elements),
+                tuple_constructor=lambda *elements: elements,
             )
 
-            # allocate temporary for all tuple elements
-            def allocate_temporary(tmp_name: str, dtype: ts.ScalarType):
-                declarations.append(itir.Temporary(id=tmp_name, domain=domain_expr, dtype=dtype))  # noqa: B023 # function only used inside loop
+            tmp_domains: SymbolicDomain | tuple[SymbolicDomain | tuple, ...] = tmp_expr.annex.domain
```

Comment:
It's a bit weird that something called `tmp_domains` in plural might contain a single value. I would expect it to always be a collection, even if ti were empty or only had a single element. 

```suggestion
            tmp_domains: SymbolicDomain | tuple[SymbolicDomain | tuple, ...] = tmp_expr.annex.domain
```